src/hooks/useTitle/useTitle.ts

```diff
@@ -0,0 +1,19 @@
+import { useRef, useEffect } from 'react';
+
+const useTitle = (title: string, resetAfter?: boolean) => {
+  const beforeTitle = useRef<string>(document.title);
+
+  useEffect(() => {
+    document.title = title;
+  }, [title]);
+
+  useEffect(() => {
+    return () => {
+      if (resetAfter) {
+        document.title = beforeTitle.current;
+      }
+    };
+  }, []);
+};
+
+export default typeof document === 'undefined' ? () => {} : useTitle;
```

Comment:
I would just check if typeof window !=== 'undefined', it that's true than document will be there as well.

I'm tempted to go for a solution where would just define the old title oustide of the useTitle scope. useEffect should only run in the browswer in this case so there window will be defined.
```
let docTitle as string | undefined;

if (typeof window !== 'undefined') {
	docTitle = document.title;
}
```